deps/v8/src/objects.cc

```diff
@@ -7330,7 +7330,13 @@ namespace {
 
 Maybe<bool> GetPropertyDescriptorWithInterceptor(LookupIterator* it,
                                                  PropertyDescriptor* desc) {
-  if (it->state() == LookupIterator::INTERCEPTOR) {
+  bool has_access = true;
+  if (it->state() == LookupIterator::ACCESS_CHECK) {
+    has_access = it->HasAccess() || JSObject::AllCanRead(it);
+    it->Next();
```

Comment:
Is this sound?  I see `LookupIterator::Next()` has a `DCHECK_NE(JSPROXY, state_)` but if I read `JSObject::AllCanRead()` right, it can stop at a JSPROXY instance.  It seems like the logic should be:
```c++
if (has_access) it->Next();
```